win/vncconfig/Authentication.h

```diff
@@ -76,6 +96,37 @@ namespace rfb {
           regKey.setBinary(_T("Password"), 0, 0);
         }
 
+#ifdef HAVE_GNUTLS
+        if (isItemChecked(IDC_ENC_X509)) {
+          gnutls_certificate_credentials_t xcred;
+          CHECK(gnutls_global_init());
+          CHECK(gnutls_certificate_allocate_credentials(&xcred));
+          int ret = gnutls_certificate_set_x509_key_file (xcred,
+                                                          regKey.getString("X509Cert"),
+                                                          regKey.getString("X509Key"),
+                                                          GNUTLS_X509_FMT_PEM);
```

Comment:
Shouldn't this be handled in `SSecurityTLS`? Seems like duplication.

Reply:
I was trying to add a sanity check to verify that the file selected is actually an X.509 cert, that the key matches, etc. before commiting the values to the registry.  Otherwise, I'm not sure that the user will get useful debug info in the event that GnuTLS fails to initialize the SSL context in SSecurityTLS.  I don't have any objection to moving those checks into SSecurityTLS or whatever...

Reply:
I think we should move that in to the core somehow. If nothing else you are assuming that `SSecurityTLS` is implemented using GnuTLS by doing this. :)

Reply:
OK.  I'll take a look and see where it makes sense to move it to.

Reply:
Ping

Reply:
You're right, this can be implemented cleaner in `SSecurityTLS`, and the additional benefit is more concise error reporting for Xvnc too.  I'll have an updated pull request later today.